### DISCUSSION TITLE
Intergration of Azure AD authentication to Wiki.js

### DISCUSSION THREAD

**Discussion by Srsnik** [requarks/wiki#4094] [Help / Questions]
So, we have this problem with our Wiki authentication. We use Azure AD as our authentication method, so that our employees can login easily but when they try to logout from the wiki it won’t sign them out. They stay signed in until they either delete cookies or restart the browser. The interesting part is that when they sign-in to the Wiki they can sign-in to Microsoft apps (e.g., Office 365) as well. However, when they sign out of the Microsoft app or page, now it will sign them out for real from the Wiki as well. The problem probably is that when signing out of the Wiki, it probably doesn’t clear the Microsoft cookies. Does anyone have a similar problem or some solution for this?

We tried those solutions but nothing worked or at least I don't know how to integrate them into our system:

https://stackoverflow.com/questions/32592787/owin-authentication-signout-doesnt-remove-cookies
https://stackoverflow.com/questions/39386633/how-to-signout-from-an-azure-application?answertab=active#tab-top

**Comment by argentina-sergio-cordoba** [comment]
cancelar suscripcion


El jue, 17 jun 2021 a las 8:45, Srsnik Peter ***@***.***>)
escribió:

> So, we have this problem with our Wiki authentication. We use Azure AD as
> our authentication method, so that our employees can login easily but when
> they try to logout from the wiki it won’t sign them out. They stay signed
> in until they either delete cookies or restart the browser. The interesting
> part is that when they sign-in to the Wiki they can sign-in to Microsoft
> apps (e.g., Office 365) as well. However, when they sign out of the
> Microsoft app or page, now it will sign them out for real from the Wiki as
> well. The problem probably is that when signing out of the Wiki, it
> probably doesn’t clear the Microsoft cookies. Does anyone have a similar
> problem or some solution for this?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Requarks/wiki/discussions/4094>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS4O2PDMWDG2EVHDW7JBRXDTTHN4JANCNFSM463P3ZAQ>
> .
>


-- 
sergio picco

**Comment by amal-viswa** [comment]
Hey, i'm having trouble while setting up azure ad authentication for wiki.js , I've followed the wiki.js official documentation for azure ad , but still not able to login error . can you guys help me resolve it

error 
![Screenshot 2022-08-23 at 18-12-26 Error](https://user-images.githubusercontent.com/90927394/186366847-003c2c5d-a09c-4752-912e-c43e10bfe999.png)
-
